### PR TITLE
Dop/display scene

### PR DESCRIPTION
**Pull request by doneill**
@shellygill Please review this initial 3D implementation.

### CODE REVIEW

File: display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+public class MainActivity extends AppCompatActivity {
+
+    private SceneView mSceneView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        // inflate MapView from layout
```
Comment (shellygill) [inline display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java]: Typo - inflate SceneView

File: display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import android.support.v7.app.AppCompatActivity;
+
+import com.esri.arcgisruntime.mapping.ArcGISScene;
+import com.esri.arcgisruntime.mapping.ArcGISTiledElevationSource;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Surface;
+import com.esri.arcgisruntime.mapping.view.Camera;
+import com.esri.arcgisruntime.mapping.view.SceneView;
+
+import static com.esri.arcgisruntime.sample.displayscene.R.id.sceneView;
```
Comment (shellygill) [inline display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java]: This is a different pattern to that which we typically use - no specific problem with it, just wondered if it was deliberate?

File: display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        // add a camera and initial camera position
+        Camera camera = new Camera(28.4, 83.9, 10010.0, 10.0, 80.0, 300.0);
+        mSceneView.setViewpointCamera(camera);
+    }
+
+    @Override
+    protected void onPause(){
+        super.onPause();
+        // pause MapView
```
Comment (shellygill) [inline display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java]: Typo - SceneView, same in onResume below.

File: display-scene/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Display a Scene
+This sample demonstrates how to display an scene with elevation data.
```
Comment (shellygill) [inline display-scene/README.md]: typo - how to display *a* scene

File: display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        ArcGISScene agsScene = new ArcGISScene();
+        agsScene.setBasemap(Basemap.createImagery());
+        mSceneView.setScene(agsScene);
+
+        // add base surface for elevation data
+        ArcGISTiledElevationSource elevationSource = new ArcGISTiledElevationSource(
+                getResources().getString(R.string.elevation_image_service));
+        Surface surface = new Surface();
+        surface.getElevationSources().add(elevationSource);
+        agsScene.setBaseSurface(surface);
```
Comment (shellygill) [inline display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java]: A new ArcGISScene already has a Surface, so you can just add the new elevation source to the existing surface.

File: display-scene/src/main/res/values/colors_blue_green.xml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <color name="colorPrimary">#2196F3</color>
+  <color name="colorPrimaryDark">#1976D2</color>
+  <color name="colorPrimaryLight">#BBDEFB</color>
+  <color name="colorAccent">#4CAF50</color>
+  <color name="colorPrimaryText">#212121</color>
+  <color name="colorSecondarytext">#757575</color>
+  <color name="colorIcons">#FFFFFF</color>
+  <color name="colorDivider">#BDBDBD</color>
```
Comment (shellygill) [inline display-scene/src/main/res/values/colors_blue_green.xml]: Are the values other than colorPrimary, colorPrimaryDark and colorAccent actually used?

File: display-scene/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+## Features
+* ArcGISScene
+* ArcGISTiledElevationSource
+* SceneView
+
+## Developer Pattern
+Create an `ArcGISScene` and set the `Basemap` with `ArcGISScene.setBasemap()`. Create a `SceneView` and set the scene to the view, `SceneView.setScene(scene)`.  Create a `Surface` and add an `ArcGISTiledElevationSource`, `Surface.getElevationSources().add()`. Set the surface as the scene's base surface `ArcGIScene.setBaseSurface(surface)`.
+
+```java
+// inflate MapView from layout
```
Comment (shellygill) [inline display-scene/README.md]: Typo - SceneView

File: display-scene/src/main/res/values/colors_blue_green.xml (hunk 6 above)
Comment (doneill) [inline display-scene/src/main/res/values/colors_blue_green.xml]: No not using them all here, I am thinking about standard color template for all samples so included all defined, will reduce.

File: display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java (hunk 2 above)
Comment (doneill) [inline display-scene/src/main/java/com/esri/arcgisruntime/sample/displayscene/MainActivity.java]: Not sure why it cannot resolve the `SceneView`.

**Comment by doneill** [comment]
@shellygill Addressed your comments, not sure why the static import is required though.

File: display-scene/README.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+mSceneView = (SceneView) findViewById(sceneView);
+// create a scene and add a basemap to it
+ArcGISScene agsScene = new ArcGISScene();
+agsScene.setBasemap(Basemap.createImagery());
+mSceneView.setScene(agsScene);
+
+// add base surface for elevation data
+ArcGISTiledElevationSource elevationSource = new ArcGISTiledElevationSource(
+        getResources().getString(R.string.elevation_image_service));
+Surface surface = new Surface();
```
Comment (shellygill) [inline display-scene/README.md]: Code snippet should be updated to match source change.

**Comment by doneill** [comment]
Thanks @shellygill, misunderstood the static import comment, removed and merging :-)

**Pull request merged**